omod/src/main/java/org/openmrs/module/spa/SpaActivator.java

```diff
@@ -11,19 +11,57 @@
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
 import org.openmrs.module.BaseModuleActivator;
+import org.openmrs.module.ModuleConstants;
+import org.openmrs.module.ModuleException;
+import org.openmrs.util.OpenmrsUtil;
+
+import java.io.File;
 
 public class SpaActivator extends BaseModuleActivator {
 
 	private final Log log = LogFactory.getLog(getClass());
+	private final String FRONTENT_DIRECTORY = "frontend";
+	private final String SINGLE_SPA_STATIC_FILES_DIR = "spa.frontend.directory";
 	
 	@Override
 	public void started() {
+
+		createFrontendDirectoryInAppDirectory();
 		log.info("SPA module started");
 	}
 
 	@Override
 	public void stopped() {
 		log.info("SPA module stopped");
 	}
+
+	public void createFrontendDirectoryInAppDirectory() {
```

Comment:
Some method documentation that describes the behavior with full paths vs relative paths would be helpful

Reply:
/**
	 * Creates a directory to hold static files for the frontend.
	 * The name/path of the directory is managed through a global property.
	 * Implementations can change the directory name/path
	 */
Is that sufficient?

Reply:
The surprising behavior is the difference between full and relative paths. That's the important thing to document here, something like "if a relative path is provided, it is treated as relative to the application data directory."

Though actually it occurs to me that the single responsibility principle suggests that we should refactor the global property lookup into `started()` and have `createFrontendDirectory` take the path as an argument. But that's picking nits. The important thing is the documentation, anyway.